cityofaustin/atd-moped
# Ability to advanced search using multiple filters as "or"/"any" condition

## Associated issues
https://github.com/cityofaustin/atd-data-tech/issues/11282

This PR:
- adds an advanced search radio group to select either **and** or **or** logic
- updates `useAdvancedSearch` to handle and apply that logic to the advanced filters in the GraphQL query
- updates the code to use React Router [useSearchParams](https://reactrouter.com/en/main/hooks/use-search-params#usesearchparams) to manage the url query strings

## Testing
**URL to test:** 

https://deploy-preview-1187--atd-moped-main.netlify.app/moped/

**Steps to test:**
1. Go to the project list view in the deploy preview
2. Add an advanced filter by click the filter icon in the search box
3. Try searching for:
- **ALL**
- **Lead is COA ATD Project Delivery**
- **Sponsor is COA ATD Active Transportation & Street Design**
- Notice that there are radios for **all** and **any** above the advanced filter rows. Keep it as **all**
- You should see one result that created that is named **#11282 test project** that matches both of the filters
- Check the url and you should see two query parameters: **filter** (with the encoded filter string) and **isOr** which should have the value of **false**
4. Now, try searching for:
- **ANY**
- **Lead is COA ATD Project Delivery**
- **Sponsor is COA ATD Active Transportation & Street Design**
- Change the radios to match **any**
- You should see many results that meet either of the two filters
- Check the url and you should see two query parameters: **filter** (with the encoded filter string) and **isOr** which should have the value of **true**
5. Try combinations of advanced filters like **projects where Arterial Management is either the Lead, Sponsor, or a Partner** from the issue, and try to break it! (You might run into https://github.com/cityofaustin/atd-data-tech/issues/14495 - that fix is coming in the next PR)
6. Try copying your url with **filter** and **isOr** parameters into a different tab and the UI should reflect the advanced filters stored in the encoded string with the correct any/all option selected in the radio buttons
7. Try applying advanced filters to the view and then entering a search string into the search box and pressing enter. You should see the advanced filter UI reset and the query parameters removed.

---
#### Ship list
